source/extensions/http/header_validators/envoy_default/header_validator.cc

```diff
@@ -584,6 +586,37 @@ void HeaderValidator::sanitizePathWithFragment(::Envoy::Http::RequestHeaderMap&
   }
 }
 
+PathNormalizer::PathNormalizationResult
+HeaderValidator::sanitizeEncodedSlashes(::Envoy::Http::RequestHeaderMap& header_map) {
+  if (!header_map.Path()) {
+    return PathNormalizer::PathNormalizationResult::success();
+  }
+  const auto escaped_slashes_action =
+      config_.uri_path_normalization_options().path_with_escaped_slashes_action();
+
+  if (escaped_slashes_action !=
```

Comment:
nit: Let's use early returns here to reduce nesting:

```
  if (escaped_slashes_action !=
      HeaderValidatorConfig::UriPathNormalizationOptions::KEEP_UNCHANGED) {
    return PathNormalizer::PathNormalizationResult::success();
  }
  // When path normalization is enabled decoding of slashes is done as part of the normalization
  // function for performance.
  auto escaped_slashes_result = PathUtil::unescapeSlashes(header_map);
  if (escaped_slashes_result != PathUtil::UnescapeSlashesResult::FoundAndUnescaped) {
    return PathNormalizer::PathNormalizationResult::success();
  }
  if (escaped_slashes_action ==
      HeaderValidatorConfig::UriPathNormalizationOptions::REJECT_REQUEST) {
    return {PathNormalizer::PathNormalizationResult::Action::Reject,
                UhvResponseCodeDetail::get().EscapedSlashesInPath};
  }
  if (escaped_slashes_action ==
                 HeaderValidatorConfig::UriPathNormalizationOptions::UNESCAPE_AND_REDIRECT) {
    return {PathNormalizer::PathNormalizationResult::Action::Redirect,
                ::Envoy::Http::PathNormalizerResponseCodeDetail::get().RedirectNormalized};
  }
  ASSERT(escaped_slashes_action ==
               HeaderValidatorConfig::UriPathNormalizationOptions::UNESCAPE_AND_FORWARD);
   return PathNormalizer::PathNormalizationResult::success();
```

Reply:
Done